src/diffusers/loaders.py

```diff
@@ -566,4 +566,426 @@ def load_textual_inversion(
+            )
+
+        allow_pickle = False
+        if use_safetensors is None:
+            use_safetensors = is_safetensors_available()
+            allow_pickle = True
+
+        user_agent = {
+            "file_type": "attn_procs_weights",
+            "framework": "pytorch",
+        }
+
+        model_file = None
+        if not isinstance(pretrained_model_name_or_path_or_dict, dict):
+            # Let's first try to load .safetensors weights
+            if (use_safetensors and weight_name is None) or (
+                weight_name is not None and weight_name.endswith(".safetensors")
+            ):
+                try:
+                    model_file = _get_model_file(
+                        pretrained_model_name_or_path_or_dict,
+                        weights_name=weight_name or LORA_WEIGHT_NAME_SAFE,
+                        cache_dir=cache_dir,
+                        force_download=force_download,
+                        resume_download=resume_download,
+                        proxies=proxies,
+                        local_files_only=local_files_only,
+                        use_auth_token=use_auth_token,
+                        revision=revision,
+                        subfolder=subfolder,
+                        user_agent=user_agent,
+                    )
+                    state_dict = safetensors.torch.load_file(model_file, device="cpu")
+                except IOError as e:
+                    if not allow_pickle:
+                        raise e
+                    # try loading non-safetensors weights
+                    pass
+            if model_file is None:
+                model_file = _get_model_file(
+                    pretrained_model_name_or_path_or_dict,
+                    weights_name=weight_name or LORA_WEIGHT_NAME,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    resume_download=resume_download,
+                    proxies=proxies,
+                    local_files_only=local_files_only,
+                    use_auth_token=use_auth_token,
+                    revision=revision,
+                    subfolder=subfolder,
+                    user_agent=user_agent,
+                )
+                state_dict = torch.load(model_file, map_location="cpu")
+        else:
+            state_dict = pretrained_model_name_or_path_or_dict
+
+        # Load the layers corresponding to UNet.
+        if state_dict.get(self.unet_name, None) is not None:
+            logger.info(f"Loading {self.unet_name}.")
+            self.unet.load_attn_procs(state_dict[self.unet_name])
+
+        # Load the layers corresponding to text encoder and make necessary adjustments.
+        if state_dict.get(self.text_encoder_name, None) is not None:
+            logger.info(f"Loading {self.text_encoder_name}.")
+            attn_procs_text_encoder = self.load_attn_procs(state_dict[self.text_encoder_name], **kwargs)
```

Comment:
Do we need to pass all the kwargs here?
```suggestion
            attn_procs_text_encoder = self.load_attn_procs(state_dict[self.text_encoder_name])
```